### PR TITLE
elementwise max min

### PR DESCRIPTION
**Pull request by JiayiFeng**
solve #7567

### CODE REVIEW

File: paddle/operators/elementwise_max_op.h [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+namespace paddle {
+namespace operators {
+
+template <typename T>
+struct MaxFunctor {
+  inline HOSTDEVICE T operator()(T a, T b) const { return a > b ? a : b; }
+};
+
+template <typename DeviceContext, typename T>
+class ElementwiseMaxKernel : public framework::OpKernel<T> {
```
Comment (QiJune) [inline paddle/operators/elementwise_max_op.h]: It seems that Max and Min are almost the same logic. We can reuse the code. Please refer to https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/reduce_op.h#L97  and https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/operators/reduce_op.h#L253

Comment (JiayiFeng) [reply paddle/operators/elementwise_max_op.h]: Sounds great.

Comment (JiayiFeng) [reply paddle/operators/elementwise_max_op.h]: fixed

File: paddle/operators/elementwise_max_op.h [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+      auto dx_e = framework::EigenVector<T>::Flatten(*dx);
+      dx_e.device(d) = (x_e > y_e).template cast<T>() * dz_e;
+    }
+    if (dy) {
+      auto dy_e = framework::EigenVector<T>::Flatten(*dy);
+      dy_e.device(d) = (x_e <= y_e).template cast<T>() * dz_e;
+    }
+  }
+};
+
```
Comment (QiJune) [inline paddle/operators/elementwise_max_op.h]: @JiayiFeng  We can reduce more codes. But we can do it later, if we have more element-wise operators to implement. Anyway, let's find a general pattern and clean code later.
```cpp
template <typename T>
struct MaxGradFunctor1 {
  template <typename Device, typename Dxe, typename Xe, typename Ye, typename Dze>
  void operator()(Device d, Dxe dx_e, Xe, x_e, Ye, y_e, Dze, dz_e) {
    dx_e.device(d) = (x_e > y_e).template cast<T>() * dz_e;
  }
};

template <typename T>
struct MinGradFunctor1 {
  template <typename Device, typename Dxe, typename Xe, typename Ye, typename Dze>
  void operator()(Device d, Dxe dx_e, Xe, x_e, Ye, y_e, Dze, dz_e) {
    dx_e.device(d) = (x_e < y_e).template cast<T>() * dz_e;
  }
};

template <typename T, typename Functor1, typename Functor2>
struct ElementwiseMinMaxGradFunctor {
  template <typename Device, typename X, typename Y, typename Z, typename dX,
            typename dY, typename dZ>
  void operator()(Device d, X x, Y y, Z z, dX dx, dY dy, dZ dz) {
    auto x_e = framework::EigenVector<T>::Flatten(*x);
    auto y_e = framework::EigenVector<T>::Flatten(*y);
    auto dz_e = framework::EigenVector<T>::Flatten(*dz);

    if (dx) {
      auto dx_e = framework::EigenVector<T>::Flatten(*dx);
      Functor1 functor;
      functor(d, dx_e, x_e, y_e, dz_e);
    }
    if (dy) {
      auto dy_e = framework::EigenVector<T>::Flatten(*dy);
      Functor2 functor;
      functor(d, dy_e, x_e, y_e, dz_e);
    }
  }
};
```

Comment (JiayiFeng) [reply paddle/operators/elementwise_max_op.h]: @QiJune 
I agree. Find a general pattern in the backward pass is not as easy as that in forwarding pass, for backward `op` takes `Z`, `dZ`, `X`, `Y` while forwarding `op` only takes `X` and `Y`.

**Pull request merged**